src/app/COMPONENTS/task-details/task-details.component.ts

```diff
@@ -32,10 +34,35 @@ export class TaskDetailsComponent implements OnInit {
   })
 
 
+
   constructor(private tasksService: TasksService, private activatedRoute: ActivatedRoute, private fb: FormBuilder) { }
 
   ngOnInit() {
-      this.task$ =  this.activatedRoute.params.pipe(map((params) => params['id'] as string),
+    this.task$ = this.activatedRoute.params.pipe(map((params) => params['id'] as string),
       switchMap((taskId) => this.tasksService.getTaskById(taskId)))
   }
+
+  updateTaskDetails(taskId: string) {
+    this.editing = false
+    this.taskForm.disable()
+    const updatedTaskDetails = this.taskForm.value
+
+    this.task$ = this.tasksService.updateTask(updatedTaskDetails, taskId)
+  }
+  allowTaskEdit() {
+
+    this.editing = true
+
+    this.task$?.pipe(
+      take(1)
+    ).subscribe(task => {
+      this.taskForm = this.fb.group({
+        title: new FormControl({ value: task.title, disabled: false }, Validators.required),
```

Comment:
you could assign these values in ngOnInit with tap operator instead of here, this would mean that all the values initially are empty, they should be visible on page load